src/libraries/System.Threading.Channels/src/System/Collections/Generic/Deque.cs

```diff
@@ -63,6 +63,12 @@ public T DequeueHead()
             return item;
         }
 
+        public T PeekHead()
+        {
+            Debug.Assert(!IsEmpty); // caller's responsibility to make sure there are elements remaining
+            return _array[_head];
```

Comment:
after double-checking that this type is `internal` it LGTM (otherwise it would be scary)